src/EnduranceTests/Handlers/HeartbeatHandler.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace NServiceBus.AzureServiceBus.EnduranceTests.Handlers
+{
+    using System.Threading.Tasks;
+    using Commands;
+    using Logging;
+
+    public class HeartbeatHandler : IHandleMessages<Heartbeat>
+    {
+        static readonly ILog log = LogManager.GetLogger<HeartbeatHandler>();
+
+        public async Task Handle(Heartbeat message, IMessageHandlerContext context)
+        {
+            log.InfoFormat("HeartbeatHandler Heartbeat with Wait of {0}", TestSettings.Rate);
+
+            if (TestSettings.TestRunId != message.TestRunId)
```

Comment:
Not sure if this is needed... With ATTs it is crucial because outcome of the test should solely depend on the messages from the same run. This is endurance tests, so messages are compatible. As long as we receive and send, all good. I'd say we remove it.